arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -502,6 +502,70 @@ public inline fun <Error, A, B, C, D, E, F, G, H, I, J> Raise<NonEmptyList<Error
   return block(unbox(a), unbox(b), unbox(c), unbox(d), unbox(e), unbox(f), unbox(g), unbox(h), unbox(i))
 }
 
+@RaiseDSL
+public inline fun <Error, A> Raise<Error>.forEachAccumulating(
+  iterable: Iterable<A>,
+  combine: (Error, Error) -> Error,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulating(iterable.iterator(), combine, block)
+
+@RaiseDSL
+public inline fun <Error, A> Raise<Error>.forEachAccumulating(
+  sequence: Sequence<A>,
+  combine: (Error, Error) -> Error,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulating(sequence.iterator(), combine, block)
+
+@RaiseDSL
+public inline fun <Error, A> Raise<Error>.forEachAccumulating(
+  iterator: Iterator<A>,
+  combine: (Error, Error) -> Error,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = recover({
+  for (item in iterator) {
+    block(RaiseAccumulate(this), item)
+  }
+}) { firstErrors ->
+  var error = firstErrors.reduce(combine)
+  for (item in iterator) {
+    recover({
+      block(RaiseAccumulate(this), item)
+    }) {
+      error = combine(error, it.reduce(combine))
+    }
+  }
+  raise(error)
+}
+
+@RaiseDSL
+public inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
+  iterable: Iterable<A>,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulating(iterable.iterator(), block)
+
+@RaiseDSL
+public inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
+  sequence: Sequence<A>,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = forEachAccumulating(sequence.iterator(), block)
+
+@RaiseDSL
+public inline fun <Error, A> Raise<NonEmptyList<Error>>.forEachAccumulating(
+  iterator: Iterator<A>,
+  @BuilderInference block: RaiseAccumulate<Error>.(A) -> Unit
+): Unit = recover({
+  for (item in iterator) {
+    block(RaiseAccumulate(this), item)
+  }
+}) { firstError ->
+  buildList {
+    addAll(firstError)
+    for (item in iterator) {
+      recover({ block(RaiseAccumulate(this), item) }) { addAll(it) }
+    }
+  }.toNonEmptyListOrNull()!!.let(::raise)
+}
```

Comment:
Same as above, I would prefer to keep the simpler version without so many nested `recover`

Reply:
It looks nicer with `attempt` btw :). I had my original implementation with that. It worked nicely because the error path is more complicated than the non-error one